pkg/api/controllers/host.go

```diff
@@ -0,0 +1,179 @@
+// Copyright 2019 The OpenSDS Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/*
+This module implements a entry into the OpenSDS northbound service.
+
+*/
+
+package controllers
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/opensds/opensds/pkg/api/policy"
+	c "github.com/opensds/opensds/pkg/context"
+	"github.com/opensds/opensds/pkg/db"
+	"github.com/opensds/opensds/pkg/model"
+)
+
+type HostPortal struct {
+	BasePortal
+}
+
+func NewHostPortal() *HostPortal {
+	return &HostPortal{}
+}
+
+func (p *HostPortal) ListHosts() {
```

Comment:
Can we add filter for the hosts result based on some parameters like osType etc. ?

Reply:
will support it will different PR, see issue here: #1042

Reply:
Okay